src/ray/gcs/pb_utils.cc

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO(hjiang): Move all functions from `pb_utils.h` to this implementation file.
+
+#include <string_view>
+
+#include "absl/strings/str_format.h"
+#include "ray/gcs/pb_util.h"
+
+namespace ray::gcs {
+
+std::shared_ptr<ray::rpc::ErrorTableData> CreateErrorTableData(
+    const std::string &error_type,
+    const std::string &error_msg,
+    absl::Time timestamp,
+    const JobID &job_id) {
+  uint32_t max_error_msg_size_bytes = RayConfig::instance().max_error_msg_size_bytes();
+  auto error_info_ptr = std::make_shared<ray::rpc::ErrorTableData>();
+  error_info_ptr->set_type(error_type);
+  if (error_msg.length() > max_error_msg_size_bytes) {
+    std::string formatted_error_message = absl::StrFormat(
+        "The message size exceeds %d bytes. Find the full log from the log files. Here "
+        "is abstract: %s",
+        max_error_msg_size_bytes,
+        std::string_view{error_msg}.substr(0, max_error_msg_size_bytes));
+    error_info_ptr->set_error_message(std::move(formatted_error_message));
+  } else {
+    error_info_ptr->set_error_message(error_msg);
+  }
+  error_info_ptr->set_timestamp(absl::ToUnixSeconds(timestamp));
```

Comment:
this limits time resolution to seconds, which is not acceptable. let's use milliseconds, or `ToDoubleSeconds`. Either way, please annotate the proto on what unit it uses.

Reply:
> or ToDoubleSeconds

I'm not sure what do you mean here.
It's made for `absl::Duration`, instead of `absl::Time`.
https://github.com/abseil/abseil-cpp/blob/2fcebef792241578e8c9dcfbba5745f834509dbb/absl/time/time.h#L647

Reply:
Updated to millisecond as you wish, also add a comment for unit in the protobuf.
But checking the code base, if we want to update the field name, there're way too many places to update in the PR, I leave a TODO.